container/lxd/initialisation_test.go

```diff
@@ -72,3 +72,97 @@ func (s *InitialiserSuite) TestNoSeriesPackages(c *gc.C) {
 		paccmder.InstallCmd("lxd"),
 	})
 }
+
+func (s *InitialiserSuite) TestEditLXDBridgeFile(c *gc.C) {
+	input := `# WARNING: Don't modify this file by hand, it is generated by debconf!
+# To update those values, please run "dpkg-reconfigure lxd"
+
+# Whether to setup a new bridge
+USE_LXD_BRIDGE="true"
+EXISTING_BRIDGE=""
+
+# Bridge name
+LXD_BRIDGE="lxdbr0"
+
+# dnsmasq configuration path
+LXD_CONFILE=""
+
+# dnsmasq domain
+LXD_DOMAIN="lxd"
+
+# IPv4
+LXD_IPV4_ADDR="10.0.4.1"
+LXD_IPV4_NETMASK="255.255.255.0"
+LXD_IPV4_NETWORK="10.0.4.1/24"
+LXD_IPV4_DHCP_RANGE="10.0.4.2,10.0.4.100"
+LXD_IPV4_DHCP_MAX="50"
+LXD_IPV4_NAT="true"
+
+# IPv6
+LXD_IPV6_ADDR="2001:470:b2b5:9999::1"
+LXD_IPV6_MASK="64"
+LXD_IPV6_NETWORK="2001:470:b2b5:9999::1/64"
+LXD_IPV6_NAT="true"
+
+# Proxy server
+LXD_IPV6_PROXY="true"
+`
+	expected := `# WARNING: Don't modify this file by hand, it is generated by debconf!
+# To update those values, please run "dpkg-reconfigure lxd"
+
+# Whether to setup a new bridge
+USE_LXD_BRIDGE="true"
+EXISTING_BRIDGE=""
+
+# Bridge name
+LXD_BRIDGE="lxdbr0"
+
+# dnsmasq configuration path
+LXD_CONFILE=""
+
+# dnsmasq domain
+LXD_DOMAIN="lxd"
+
+# IPv4
+LXD_IPV4_ADDR="10.0.19.1"
+LXD_IPV4_NETMASK="255.255.255.0"
+LXD_IPV4_NETWORK="10.0.19.1/24"
+LXD_IPV4_DHCP_RANGE="10.0.19.2,10.0.19.254"
+LXD_IPV4_DHCP_MAX="253"
+LXD_IPV4_NAT="true"
+
+# IPv6
+LXD_IPV6_ADDR="2001:470:b2b5:9999::1"
+LXD_IPV6_MASK="64"
+LXD_IPV6_NETWORK="2001:470:b2b5:9999::1/64"
+LXD_IPV6_NAT="true"
+
+# Proxy server
+LXD_IPV6_PROXY="false"
+
+`
+	result := editLXDBridgeFile(input, "19")
+	c.Assert(result, jc.DeepEquals, expected)
+}
+
+func (s *InitialiserSuite) TestDetectSubnet(c *gc.C) {
+	input := `    inet 127.0.0.1/8 scope host lo
+    inet6 ::1/128 scope host 
+    inet 192.168.0.69/24 brd 192.168.0.255 scope global eth0
+    inet6 fd5d:e5bb:c5f9::c0c/128 scope global dynamic 
+    inet6 fd5d:e5bb:c5f9:0:1e6f:65ff:fed5:5698/64 scope global noprefixroute dynamic 
+    inet6 fe80::1e6f:65ff:fed5:5698/64 scope link 
+    inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
+    inet 192.168.100.1/24 brd 192.168.100.255 scope global virbr1
+    inet 10.0.3.1/24 scope global lxcbr0
+    inet6 fe80::a00f:35ff:fe81:f7ed/64 scope link 
+    inet6 fe80::fcd3:9dff:fee4:ba90/64 scope link 
+    inet6 fe80::fc54:ff:fe6e:2d7d/64 scope link 
+    inet6 fe80::fc54:ff:fe3e:8018/64 scope link 
+    inet6 fe80::fc54:ff:feee:c795/64 scope link 
+    inet6 fe80::fc54:ff:fe30:9216/64 scope link 
+`
+	result, err := detectSubnet(input)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, jc.DeepEquals, "4")
+}
```

Comment:
This is my current 'ip addr show' with a bunch of extra subnets for testing Juju space support:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether d4:be:d9:8e:6b:0f brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.104/24 brd 192.168.0.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::d6be:d9ff:fe8e:6b0f/64 scope link 
       valid_lft forever preferred_lft forever
3: lxcbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether fe:03:cb:4d:25:d5 brd ff:ff:ff:ff:ff:ff
    inet 10.0.3.1/24 scope global lxcbr0
       valid_lft forever preferred_lft forever
    inet6 fe80::9055:55ff:feac:1266/64 scope link 
       valid_lft forever preferred_lft forever
4: virbr0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 6e:cc:77:50:86:ca brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
       valid_lft forever preferred_lft forever
5: virbr2: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 52:54:00:36:5a:2b brd ff:ff:ff:ff:ff:ff
    inet 10.10.100.1/24 brd 10.10.100.255 scope global virbr2
       valid_lft forever preferred_lft forever
6: virbr2-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc pfifo_fast master virbr2 state DOWN group default qlen 500
    link/ether 52:54:00:36:5a:2b brd ff:ff:ff:ff:ff:ff
7: virbr1: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 52:54:00:2f:f6:02 brd ff:ff:ff:ff:ff:ff
    inet 10.10.0.1/24 brd 10.10.0.255 scope global virbr1
       valid_lft forever preferred_lft forever
8: virbr1-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc pfifo_fast master virbr1 state DOWN group default qlen 500
    link/ether 52:54:00:2f:f6:02 brd ff:ff:ff:ff:ff:ff
9: virbr3: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 52:54:00:4c:3c:09 brd ff:ff:ff:ff:ff:ff
    inet 10.10.200.1/24 brd 10.10.200.255 scope global virbr3
       valid_lft forever preferred_lft forever
10: virbr3-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc pfifo_fast master virbr3 state DOWN group default qlen 500
    link/ether 52:54:00:4c:3c:09 brd ff:ff:ff:ff:ff:ff
12: vethSGOPFC: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master lxcbr0 state UP group default qlen 1000
    link/ether fe:03:cb:4d:25:d5 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc03:cbff:fe4d:25d5/64 scope link 
       valid_lft forever preferred_lft forever
14: vethKWD9G8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master lxcbr0 state UP group default qlen 1000
    link/ether fe:53:26:4f:b9:2e brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc53:26ff:fe4f:b92e/64 scope link 
       valid_lft forever preferred_lft forever
```

It would be good on the test to include the extra lines since we want to make sure they don't confuse the parser.